lunchbox/persistentMap.h

```
@@ -79,6 +93,24 @@ class PersistentMap : public boost::noncopyable
     LUNCHBOX_API static bool handles( const servus::URI& uri );
 
     /**
+     * Create a map which can be used for caching IO on the local system.
+     *
+     * The concrete implementation used depends on the system setup and
+     * available backend implementations. If no suitable implementation is
+     * found, a null pointer is returned.
+     *
+     * The current implementation returns:
+     * * A memcached-backed cache if libmemcached is available and the
+     *   environment variable MEMCACHED_SERVERS is set (see constructor
+     *   documentation for details).
+     * * A leveldb-backed cache if leveldb is available and LEVELDB_CACHE is set
+     *   to the path for the leveldb storage.
+     *
+     * @return a PersistentMap for caching IO, or 0.
```

throw instead of nullptr?